packages/grafana-data/src/types/logs.ts

```diff
@@ -263,3 +263,42 @@ export const hasLogsContextUiSupport = (datasource: unknown): datasource is Data
 
   return withLogsSupport.getLogRowContextUi !== undefined;
 };
+
+export interface AnalyzeQueryOptions {
+  check: string;
+  attributes: KeyValue<string>;
+}
+
+/**
+ * Data sources that support query manipulation through `modifyQuery` and `analyzeQuery` in Explore.
+ * Allows for interactions such as changing the query from logs details, or displaying filter status.
+ * @internal
+ */
+export interface DataSourceToggleableQueryFiltersSupport<TQuery extends DataQuery, TAnalyzeQueryResult = boolean> {
+  /**
+   * Modify the query with a given action.
+   * To support toggleable filters, `ADD_FILTER` or `ADD_FILTER_OUT` should be implemented.
+   * If the filter is already present, it should be removed.
+   * If the opposite filter is present, it should be replaceted.
+   */
+  modifyQuery(query: TQuery, action: QueryFixAction): TQuery;
```

Comment:
We talked about renaming this function to something else than `modifyQuery` so we have deterministic API - `modifyQuery` function adds/removes filters and `toggleLabelFilter` (or something else toggles filters).

Reply:
I think specific clear API for toggling with a method that is not used for anything else would be more clear for plugin developers. 

Also now the implementation in Loki and ES is buggy cause you are using `ADD_FILTER` API, but in some cases you are removing filter. `ADD_FILTER` should always add filter in my opinion.

Reply:
Done ✨